src/v/cloud_storage/remote_partition.cc

```diff
@@ -781,64 +664,42 @@ remote_partition::timequery(storage::timequery_config cfg) {
     }
 }
 
-remote_partition::iterator remote_partition::begin() {
-    if (_segments.empty()) {
-        return end();
-    }
-    auto it = _segments.begin();
-    return iterator(_segments, it->first);
-}
-
-remote_partition::iterator remote_partition::end() {
-    return iterator(_segments);
-}
-
-remote_partition::iterator remote_partition::upper_bound(kafka::offset o) {
-    auto it = _segments.upper_bound(o);
-    if (it != _segments.end()) {
-        return iterator(_segments, it->first);
-    }
-    return end();
-}
-
 remote_partition::iterator
 remote_partition::seek_by_timestamp(model::timestamp t) {
     auto segment_meta = _manifest.timequery(t);
 
     if (segment_meta) {
-        kafka::offset o = get_kafka_base_offset(*segment_meta);
-        auto found = _segments.find(o);
+        auto found = _segments.find(segment_meta->get().base_offset);
 
         // Cannot happen because timequery() calls
         // update_segments_incrementally first.
```

Comment:
not true anymore, right? so why is the assert safe?

Reply:
fixed that